gapic/templates/%namespace/%name_%version/%sub/services/%service/async_client.py.j2

```diff
@@ -263,7 +270,7 @@ class {{ service.async_client_name }}:
             {% if not method.server_streaming %}
             ) -> {{ method.client_output_async.ident }}:
             {% else %}
-            ) -> Awaitable[AsyncIterable[{{ method.client_output_async.ident }}]]:
+            ) -> Union[Awaitable[AsyncIterable[{{ method.client_output_async.ident }}]], Awaitable[GrpcAsyncStream[{{ method.client_output_async.ident }}]]]:
```

Comment:
same comment here: it would be great if we could avoid the Union return

Reply:
I don't see any way to avoid this as this is also the response type for REST. Let's decide on the next steps for https://github.com/googleapis/gapic-generator-python/pull/2256#discussion_r1851135593 and the same solution could be applied to async.